NetworkingComparison/RxSwift/RxSwiftAPI.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  RxSwiftAPI.swift
+//  NetworkingComparison
+//
+//  Created by Ian Luo on 11/18/19.
+//  Copyright © 2019 Lokae. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxCocoa
+
+class RxSwiftAPI {
+    static let shared = RxSwiftAPI()
+
+    enum RxSwiftError: Error {
+        case badUrl, invalidKey, cityNotFound, serverFailure
+    }
+
+    private let session = URLSession.shared
+
+    private enum SessionStyle {
+        case observableRequest, standardRequest, manualResponse
+    }
+
+    private var sessionStyle: SessionStyle = .observableRequest
+
+    private init() { }
+
+    func getForecasts() -> Observable<[CodableForecast]> {
+        var data: Observable<Data>
+        switch sessionStyle {
```

Comment:
I generally prefer to create a function that returns the value I'm looking for with switch statements like this. It avoids having a mutable variable altogether and makes it easier to read (IMO).

For example:

```swift
let data = fetch(for: sessionStyle)
...


func fetch(for style: SessionStyle) -> Observable<Data> {
    switch style {
    case .observableRequest:
        return fetchWithStandardRequest()
    case .standardRequest:
        return fetchWithObservableRequest()
    case .manualResponse:
        return fetchWithManualResponse()
    }
}
```

Reply:
I like it